syntax = "proto3";
package paxoslib;
message Proposal {
  uint64 id = 1;
  bytes value = 2;
}
message MessagePrepare{
  uint64 proposal_id = 1;
}
message MessagePromised{
  bool is_accepted = 1;
  Proposal accepted_proposal = 2;
}
message MessageRejectPromise{
  uint64 promised_proposal_id = 1;
}
message MessageAccept{
  Proposal proposal = 1;
}
message MessageAccepted{
  Proposal proposal = 1;
}
message MessageRejectAccept{

}
message Message{
  enum Type{
    DEFAULT = 0;
    PREAPRE = 1;
    PROMISED = 2;
    REJECT_PREOMISE = 3;
    ACCEPT = 4;
    ACCEPTED = 5;
    REJECT_ACCEPT = 6;
  }
  Type type = 1;
  uint32 from_node_id = 2;
  uint32 to_node_id = 3;
  uint64 instance_id = 4;
  uint64 id = 5;
  oneof message_oneof {
    MessagePrepare prepare = 101;
    MessagePromised promised = 102;
    MessageRejectPromise reject_promise = 103;
    MessageAccept accept = 104;
    MessageAccepted accepted = 105;
    MessageRejectAccept reject_accept = 106;
  }
  

}